### PR TITLE
Checkout: add ability to toggle summary on mobile

### PR DESCRIPTION
**Pull request by michaeldcain**
Currently, the checkout summary (with plan features, cost summary, and happiness callout) is hidden on mobile devices. This allows it to be toggled on mobile devices, behind the "purchase details" header.

Changes from the design:
- I used `MaterialIcon` arrow component to match the sidebar toggle pattern in Calypso
- I left the cart total visible. I think this information will be valuable (especially with tax and coupon items) in steps after the review step

Collapsed | Expanded
------------ | -------------
mobile-collapsed | mobile-expanded
tablet-collapsed | tablet-expanded

**To test:**
- view Checkout with a viewport `<960px`
- verify that the "Purchase Details" header is visible with the current cart total
- verify that the checkout summary is collapsed by default
- verify that clicking on the "Purchase Details" header toggles the checkout summary
- verify that in a viewport `>960px` the header is hidden and the summary is permanently visible (floated to the side)

### CODE REVIEW

**Comment by fditrapani** [comment]
Nice work! That big tap area on the toggle feels great — you can't miss it! 

I found something small with the borders when there's an up sell on the between-mobile-and-desktop view:

image.

Looking on mobile, the end-to-end line makes the up sell look blend into the rest of the screen. What if we separate it with a line like this:

image

I noticed the info icon next to the "Purchase Details" title. It adds a little of that new balance flavour :) We were talking about adding a notification to this title when there's an up sell in the cart. Are you planning on still giving that a try? or is that for another PR?

**Comment by michaeldcain** [comment]
@fditrapani: I think it's ready for one more set of 👀

File: client/my-sites/checkout/composite-checkout/wpcom/components/wp-checkout.js [hunk 1]
```diff
@@ -240,28 +257,107 @@ export default function WPCheckout( {
...
-const CheckoutSummaryUI = styled( CheckoutSummaryArea )`
+const CheckoutSummaryTitleLink = styled.button`
+	background: ${( props ) => props.theme.colors.background};
+	border-bottom: 1px solid ${( props ) => props.theme.colors.borderColorLight};
+	color: ${( props ) => props.theme.colors.textColor};
+	display: flex;
+	font-size: 16px;
+	font-weight: ${( props ) => props.theme.weights.bold};
+	justify-content: space-between;
+	padding: 20px;
```
Comment (fditrapani) [inline client/my-sites/checkout/composite-checkout/wpcom/components/wp-checkout.js:268]: Sorry, I missed this small alignment issue in my previous review. Can we update it to look like the screen on the right?

image

I believe the right measurements are `padding: 20px 23px 20px 14px`

**Comment by fditrapani** [comment]
Other than one minor alignment issue, these updates look great. Thank you!

**Comment by fditrapani** [comment]
The updated padding looks good! @michaeldcain! 

> The only thing I have to note is that the upsell margin seems to be slightly off compared to the rest of the frame.

Good catch @sirbrillig — I saw this too. It looks like it's still rendering the left and right borders on mobile.

**Comment by michaeldcain** [comment]
@fditrapani @sirbrillig: I updated the borders for devices between phone and tablet. I think we lost those borders when I originally introduced the `CheckoutSummary` a few weeks ago.

**Comment by fditrapani** [comment]
Nice! Looks great 🚢

**Pull request merged**